db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromVTI.java

```diff
@@ -245,6 +245,12 @@ public CostEstimate estimateCost(
                 costEstimate.setSingleScanRowCount(estimatedRowCount);
                 costEstimate.setLocalCost(estimatedCost);
                 costEstimate.setRemoteCost(estimatedCost);
+                int numPartitions = costEstimate.partitionCount();
+                if (numPartitions <= 0)
+                    numPartitions = 1;
```

Comment:
Same code? Again? Please refactor to use a method that does the calculation. I'm wondering if `partitionCount` could not return `1` always. What would happen?

Reply:
This is defensive code.  Refactored as mentioned above.